### PR TITLE
modify way to handle error.

### PR DESCRIPTION
**Pull request by lyw0149**
1. sync up way to handle error with eval framework,
2.Adjust restful http status.
@ethkim @nlpenguin  @ifndefdeadmau5 @djflexible @SarahhhC @KaylaKwon

### CODE REVIEW

File: src/server/config/json/match.callback.js [hunk 1]
```diff
@@ -1,14 +1,8 @@
...
-  successSendMail: {
-    successCode: 1,
-    msg: 'send Email successfully.',
-  },
+  successSendMail: 'SendMail - Success to send an email.',
+  successResponse: 'ResponseMentoring - success Response.',
+  failSendMail: 'SendMail - Fail to send an email.',
+  matchAlreadyExist: 'RequestMentoring - Match already exist.',
+  cannotFoundMentor: 'RequestMentoring - Cannot found mentor.',
+  mongooseErr: 'An error occured at mongoose.',
```
Comment (SarahhhC) [inline src/server/config/json/match.callback.js]: Let's sync up! in survey callback we used capitalized letters like
` ERR_GET_SURVEY: 'Survey Not Found - No Matching survey_id'`.
and please consider sorting~

File: src/server/config/json/user.callback.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export default {
+  successRegister: 'Register & sign in success.',
+  successSignin: 'Sign in success .',
+  mongooseErr: 'An error occured at mongoose.',
+  failRegister: 'Failed to register',
+  failSignin: 'Failed to sign in.',
+  failAuth: 'Authentication failed. Please sign in first.',
+  invalidAccessToken: 'invalidAccessToken',
+  invalidPlatform: 'invalidPlatform',
+};
```
Comment (SarahhhC) [inline src/server/config/json/user.callback.js]: ditto

**Comment by ifndefdeadmau5** [comment]
:shipit: Thanks yangwoo!

**Comment by heartjigi** [comment]
LGTM

**Comment by heartjigi** [comment]
LGTM

**Comment by ochanje210** [comment]
call me dabin

**Pull request merged**